panel/src/panel/content.js

```diff
@@ -149,6 +159,30 @@ export default (panel) => {
 		 */
 		get values() {
 			return panel.view.props.content;
+		},
+
+		rescue() {
+			if (this.isSaved === false) {
+				const data = new FormData();
+
+				for (const field in panel.view.props.content) {
+					data.append(field, panel.view.props.content[field]);
+				}
+
+				navigator.sendBeacon(panel.url(this.api + "/changes/save"), data);
+
+				// update the last modification timestamp
+				panel.view.props.lock.modified = new Date();
+				this.isSaved = true;
+			}
 		}
 	});
+
+	document.addEventListener("visibilitychange", () => {
```

Comment:
`Suggestion:` We could maybe try to move this to the events.js to make sure that we have a better overview of global event handlers.